src/Controller/CheckinRequestController.php

```diff
@@ -128,9 +175,11 @@ protected function initiateCheckinRequest(CheckinRequest $checkinRequest)
         $checkinRequest->create();
 
         // Initiate a job for non-cancellation requests.
-        if (is_null($checkinRequest->getJobId()) && $this->isUseJobService()) {
+        if (is_null($checkinRequest->getCancelRequestId()) && $this->isUseJobService()) {
+            // Create a jobId and assign it to special checkinJobId property
+            // (to signal that job should not be completed later in execution)
             $checkinRequest->setCheckinJobId(JobService::generateJobId($this->isUseJobService()));
-            // Set jobId for proper responses for non-cancellation requests.
+            // Copy jobId to jobId property (which is saved in db):
```

Comment:
What I don't understand is why if we had another service handling checkins, we wouldn't ask it to pass a `checkInId`. Then we wouldn't need to assume that the absence of a `cancelId` indicates that this is a checkin, and we should generate a `checkInId`.